lib/common/grpc-service.js

```diff
@@ -457,11 +458,12 @@ GrpcService.createDeadline_ = function(timeout) {
 GrpcService.getError_ = function(err) {
   if (err && GRPC_ERROR_CODE_TO_HTTP[err.code]) {
     var defaultErrorDetails = GRPC_ERROR_CODE_TO_HTTP[err.code];
-    err.code = defaultErrorDetails.code;
-    err.message = err.message || defaultErrorDetails.message;
-    return err;
-  }
 
+    return extend(true, new Error(), err, {
```

Comment:
I think we just want to be sure:

- We return an `Error` object
- The `.stack` property is useful

This probably assigns a new `stack` to this location in the code, which I suppose is correct?